mlir/lib/Dialect/Affine/Transforms/LoopFusion.cpp

```diff
@@ -237,6 +299,58 @@ static void sinkSequentialLoops(MemRefDependenceGraph::Node *node) {
   node->op = newRootForOp;
 }
 
+/// Creates a private memref to be used by vector operations.
+/// TODO: The difference between this and 'createPrivateMemRef' is that
+/// the system for calculating the bounds and constraints doesn't
+/// support vector operations. Thus, we use the shape of the vector
```

Comment:
But this looks wrong. The vector shape can't be the shape of the private memref. We'll have to use the constraint system. You aren't assuming single element private memref, right?